x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1203,44 +1205,51 @@ export class RulesClient {
     }
 
     if (this.eventLogger && attributes.scheduledTaskId) {
-      const { state } = taskInstanceToAlertTaskInstance(
-        await this.taskManager.get(attributes.scheduledTaskId),
-        attributes as unknown as SanitizedAlert
-      );
+      try {
```

Comment:
Wrapped this in a try/catch because I noticed as I was trying to get the functional tests to work that if the `taskManager.get` threw an error, the entire `disable` function would fail. I don't think that adding these placeholder recovery events should block disabling the rule but I'm happy to undo this change if others disagree.

Reply:
>I don't think that adding these placeholder recovery events should block disabling the rule but I'm happy to undo this change if others disagree.

Does this mean that those alert instances would get stuck in a "never recovering" state though? 🤔

Reply:
If we can't disable a rule because for some reason its associated task document is missing, then we can never re-enable it to make sure a new task doc gets scheduled for it. This seems worse to me than getting stuck in never recovering?

Reply:
Oh, I see... I didn't think about that edge case.
I'm worried about orphaning those alerts though...

Can this be addressed by handling the failure differently if it's a `404` (task doc is missing) than if it's a `500` or something else that might just be an intermittent connectivity issue?

Reply:
I think I will make a separate issue for this in order to have a more detailed discussion about what the right behavior should be. I only saw this behavior bc I wasn't setting up my functional test correctly, so not sure how frequently this might happen.

Reply:
https://github.com/elastic/kibana/issues/118024

Reply:
Thanks Ying :)